apollo-gradle-plugin-incubating/src/main/kotlin/com/apollographql/apollo/gradle/ApolloCodegenTask.kt

```diff
@@ -58,6 +55,9 @@ open class ApolloCodegenTask : SourceTask() {
     return super.getSource()
   }
 
+  @Internal
+  var schemaFile: File? = null
```

Comment:
sanity check, I assume that `schemaFile`  might come from the configuration. If it's marked as internal will this task be NOT UP-TO-DATE if user changes in plugin configuration location to schema?

Reply:
The magic comes from `apolloCodeGenTask.source(schemaFile)` there: https://github.com/martinbonnin/apollo-android/blob/7a74704161a8c21f4e6acd261a9aa937118ecede/apollo-gradle-plugin-incubating/src/main/kotlin/com/apollographql/apollo/gradle/ApolloPlugin.kt#L191.

Since the schemaFile is added to the source files, it's taken into account for up-to-date checks. 

Something I wonder is how Gradle treats `Input` properties. Would it just check the path ? Or also modification date, possibly content ? I'll look that up but for now I assumed that having the schema as source does the job. And the test passes so it should be all good.

Moving forward, I'm not sure we need ApolloCodeGenTask to inherit from `SourceTask`. Maybe it can just be a `DefaultTask` if we declare inputs correctly.